### ISSUE TITLE
Feedback docs: "500: indexed_db_went_bad"

### ISSUE THREAD

**Issue by binokaryg** [medic/cht-core#7149]
**Describe the bug**
In the month of May 2021, the feedback documents for `brac-ug` (version 3.9.2) had these errors recorded 21 times from 19 users:

Cause | Count
-- | --
`Possibly unhandled rejection: "status":500,"name":"indexed_db_went_bad","message":"unknown","error":true,"reason":"TimeoutError"}` | 13
`Possibly unhandled rejection: "status":500,"name":"indexed_db_went_bad","message":"unknown","error":true,"reason":"QuotaExceededError"}` | 8

**Logs**

Where this error appeared (based on the URLs):
URL group| Count
-- | --
Tasks page | 9
Contacts page | 4
Specific contact |  4
Targets page | 3
About page | 1
The top-level argument from the logs for all 21 of them had this message: `error: "Database has a global failure"`

**Environment**
- Instance: brac-ug.app.medicmobile.org
- Version: 3.9.2

**Additional context**
This error appears every month ranging from 4 times in Oct 2020 to 24 times in Apr 2021.

The timeout error has appeared only two times in project `pih-malawi`, and several times in `muso-app` (data not fully available yet).

**Comment by dianabarsan** [comment]
> QuotaExceededError

I'm unsure of what we could do to avoid this. 

If this is happening during indexing, it's out of our control - we don't have a mechanism to stop indexing once it's started (short of forcefully refreshing the page and not resuming by not querying any view). 

If this is happening during normal use and we could find some way of detecting that the device is running low on space (we already have something that _should_ do this in the Android app: https://github.com/medic/cht-android/blob/53b9646bb69c678b85de304e0a4d684d48e866e1/src/main/java/org/medicmobile/webapp/mobile/StartupActivity.java#L34) and alert the user. The user has to take action, though, and clear up some storage space. 

> TimeoutError

Can we get more info from other projects and see how frequently this is happening? 
Are these databases recoverable (does refreshing fix the issue) or are they permanently damaged? 
We've had reports of indexeddb failures and we even added a modal (https://github.com/medic/cht-core/issues/5045) to prompt the user to refresh their page when it happens.

**Comment by binokaryg** [comment]
The `TimeoutError` with `indexed_db_went_bad` has also been seen in other projects:
- PIH: 2 times (Nov 2020, March 2021)
- Muso (incomplete data): 51 times (29 times in Nov 2019, 1-4 times in other months until March 2021)

For BRAC, the frequency of the `TimeoutError` variant by month is:

Month | Count
-- | --
2020-09 | 1
2020-10 | 1
2020-11 | 4
2020-12 | 2
2021-01 | 3
2021-02 | 7
2021-03 | 15
2021-04 | 12
2021-05 | 13
2021-06 (current) | 3

**Comment by binokaryg** [comment]
> Are these databases recoverable (does refreshing fix the issue) or are they permanently damaged?
How do we check that?

**Comment by dianabarsan** [comment]
> > Are these databases recoverable (does refreshing fix the issue) or are they permanently damaged?
> 
> How do we check that?

By just looking at data, we could check their telemetry entries, and check the device ID. 
If the database isn't recoverable, they'd have to clear cache, which should also clear the device ID storage item. So you should see different device ids in telemetry entries for users that cleared their devices. 
Otherwise we could check if any of these users started an initial replication following an error (requires perusing API logs).

**Comment by garethbowen** [comment]
For the `QuotaExceededError` it would be really useful to know whether the users are on a Crosswalk app, a Webview app, or in a browser as each have different quotas. Some of this information is available in telemetry so we should be able to work it out.

It would also be interesting to know how many docs these users are trying to replicate.

**Comment by binokaryg** [comment]
@dianabarsan 
Out of the 13 users of BRAC who got the `TimeoutError` in May 2021, only 2 have got a new device ID in May 2021. In both cases, it was a new device model.

@garethbowen 
All the 6 users of BRAC who got `QuotaExceededError` in May 2021, their `deviceInfo` is not available, but they have:

userAgent | users
-- | --
Mozilla/5.0 (Linux; Android 6.0; TECNO-W3 Build/MRA58K; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/55.0.2883.91 Mobile Safari/537.36 | 5
Mozilla/5.0 (Linux; Android 6.0; TECNO_W4 Build/MRA58K; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/52.0.2743.98 Mobile Safari/537.36 | 1

So, it looks like the app was being accessed from browsers on the phones.

**Comment by garethbowen** [comment]
> So, it looks like the app was being accessed from browsers on the phones.

Right! And a very old version of the browser too, one of which is not supported by the CHT, and hasn't been for years. The workaround would be to use the Crosswalk version of the Android app which has unlimited storage (up to the device limit). It'd be useful to ensure that the user sees some constructive error message when this occurs.

**Comment by kennsippell** [comment]
fyi @jonathanbataire @pkitutu

**Comment by jonathanbataire** [comment]
So been in touch with BRAC and they confirmed that they've used browsers in the past but weren't sure this type of thing still happens 
@binokaryg can we get a list of these users to share with BRAC

**Comment by binokaryg** [comment]
Hi @jonathanbataire, I have shared with you via DM the list of users getting `QuotaExceededError` along with their userAgent data for the months of May and June.
